[shopsys] change forms in popup so they work with js validation (#782)
shopsys/shopsys: docs/upgrade/UPGRADE-unreleased.md

    - if you have extended `CountryController` revise your changes – `new` and `edit` actions were added
- if you have extended `Localization` class, you have to add type-hints to extended methods because they were added in the parent class ([#806](https://github.com/shopsys/shopsys/pull/806))
    - if you have extended method `Localization::getAdminLocale()` only to have administration in a different language than english, you can delete it and set parameter `shopsys.admin_locale` in your `parameters.yml` file instead

> - changed JS validation of forms in popup and login form in popup is now get by AJAX ([#782](https://github.com/shopsys/shopsys/pull/782))

Reviewer: ```suggestion
- fixed JS validation of forms in popup windows ([#782](https://github.com/shopsys/shopsys/pull/782))
    - login form in popup is now loaded via AJAX
```

>     - in `window.js` add options `textHeading = ''` and `cssClassHeading: ''` to `var defaults` and then add it to `$windowContent` and add div with `js-validation-errors` class to every popup.

Reviewer: I would change this to: 
This javascript files has been changed, please update them appropriately:
- src/Shopsys/ShopBundle/Resources/scripts/frontend/login.js
- src/Shopsys/ShopBundle/Resources/scripts/frontend/window.js

For me is better to see diff on Github than this instructions

Author: The change or login.js is mentioned in the optional point below and It is only few lines so I think it's more readable here

Reviewer: We have discussed it and you want to make it more straightforward

        - replace this:
            ```javascript
                  var $windowContent = $('<div class="js-window-content window-popup__in"></div>').html(options.content);
            ```
        - with this:
            ```javascript
                  var $windowContent = $('<div class="js-window-content window-popup__in"></div>');

                  if (options.textHeading !== '') {
                      $windowContent.append('<h2 class="' + options.cssClassHeading + '">' + options.textHeading + '</h2>');
                  }

                  $windowContent.append(
                      '<div class="display-none in-message in-message--alert js-window-validation-errors"></div>'
                      + options.content
                  );
            ```
- *(optional)* change login form so it is loaded by AJAX and works with JS validation

>     - you can change it as it was done in this [commit](https://github.com/shopsys/shopsys/pull/782/commits/bb6dadfce1d588aa3e06a8ba7b117d3de7afc1ce)

Reviewer: You did not mention changes in templates
